access-checkout/src/main/java/com/worldpay/access/checkout/client/validation/config/CardValidationConfig.kt

```diff
@@ -23,36 +35,72 @@ class CardValidationConfig private constructor(
         private var validationListener: AccessCheckoutCardValidationListener? = null
         private var lifecycleOwner: LifecycleOwner? = null
 
+        /**
+         * Sets the pan ui element
+         *
+         * @param[pan] [EditText] that represents the pan ui element
+         */
         fun pan(pan: EditText): Builder {
             this.pan = pan
             return this
         }
 
+        /**
+         * Sets the expiry date ui element
+         *
+         * @param[expiryDate] [EditText] that represents the expiry date ui element
+         */
         fun expiryDate(expiryDate: EditText): Builder {
             this.expiryDate = expiryDate
             return this
         }
 
+        /**
+         * Sets the cvc ui element
+         *
+         * @param[cvc] [EditText] that represents the cvc ui element
+         */
         fun cvc(cvc: EditText): Builder {
             this.cvc = cvc
             return this
         }
 
+        /**
+         * Sets the base url to use when calling Worldpay services
+         *
+         * @param[baseUrl] [String] that represents the base url
+         */
         fun baseUrl(baseUrl: String): Builder {
             this.baseUrl = baseUrl
             return this
         }
 
+        /**
+         * Sets the validation listener that should be notified on validation changes
+         *
+         * @param[validationListener] [AccessCheckoutCardValidationListener] that represents the validation listener
+         */
         fun validationListener(validationListener: AccessCheckoutCardValidationListener): Builder {
             this.validationListener = validationListener
             return this
         }
 
+        /**
+         * Sets the lifecycle owner of the application so that validation state can be handled during lifecycle changes
+         *
+         * @param[lifecycleOwner] [LifecycleOwner] that represents the lifecycle owner of the application
+         */
         fun lifecycleOwner(lifecycleOwner : LifecycleOwner) : Builder {
             this.lifecycleOwner = lifecycleOwner
             return this
         }
 
+        /**
+         * Builds the validation configuration by returning an instance of the [CardValidationConfig]
+         *
+         * @return [CardValidationConfig] implementation that can be used to initialise validation
+         * @throws [IllegalArgumentException] is thrown when a property is missing
```

Comment:
Do we also throw that type of exception in case where the Expiry Date is provided in an invalid format?

Reply:
This expiry date is just the ui element rather than the value, so this class does not care about the formatting. This class is used by the validation feature. You are thinking of the `CardDetails` object in the sessions.model package

Reply:
Yes, indeed!